src/ucp/wireup/ep_match.c

```diff
@@ -131,5 +131,9 @@ void ucp_ep_match_remove_ep(ucp_worker_h worker, ucp_ep_h ep)
                                UCS_CONN_MATCH_QUEUE_UNEXP :
                                UCS_CONN_MATCH_QUEUE_EXP);
 
-    ucp_ep_update_flags(ep, 0, UCP_EP_FLAG_ON_MATCH_CTX);
+    /* Release remote ID, because it is no longer needed and to make sure that
+     * flush state won't be used (flush state is considered as valid, when
+     * EP doesn't exist on matchong context and remoe EP ID is set) */
+    ucp_ep_update_flags(ep, 0,
+                        UCP_EP_FLAG_ON_MATCH_CTX | UCP_EP_FLAG_REMOTE_ID);
```

Comment:
why do we set remote id here?

Reply:
we release the flag here instead to not use flush state